nuxeo-drive-client/nxdrive/engine/engine.py

```diff
@@ -979,20 +980,38 @@ def get_user_full_name(self, userid):
         """
             Get the last contributor full name
         """
-        fullname = userid
         try:
-            if userid in self._user_cache:
-                fullname = self._user_cache[userid]
-            else:
-                rest_client = self.get_rest_api_client()
-                response = rest_client.get_user_full_name(userid)
-                if response and 'properties' in response:
-                    properties = response['properties']
-                    firstName = properties.get('firstName')
-                    lastName = properties.get('lastName')
-                    if firstName and lastName:
-                        fullname = " ".join([firstName, lastName]).strip()
-                        self._user_cache[userid] = fullname
+            rest_client = self.get_rest_api_client()
+            response = rest_client.get_user_full_name(userid)
+            if response and 'properties' in response:
+                properties = response['properties']
+                firstName = properties.get('firstName')
+                lastName = properties.get('lastName')
+                if self._dao.get_user_info(userid):
```

Comment:
see previous comment for sqlite.py.
Instead of"
if user exists:
  update
else:
  insert
you can just use:
self._dao_insert_user_info(...)